### DISCUSSION TITLE
SFTP access doesn't exist

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#4993] [Q&A] [answered]
A friend is starting a new radio and he's renting from a local reseller.  He asked me for help so that he could find his way around the dashboard and get his station up and running as smoothly as possible.  
I was telling him about SFTP access and how it was useful in building and maintaining playlists when I noticed he doesn't have such access on the Music Files page nor is there a link to set up a user under UTILITIES. I asked the reseler directly and he replied that he didn't know what SPTP was... (Found that odd...) 

Does that mean the reseller has actively blocked or failed to activate SFTP access and is there a way to activate it?

Seems to me such access was already present through no action of mine when our station was first installed...

**Comment by BusterNeece** [comment, marked as answer]
SFTP should be automatically included with any Docker installation. Ansible installations don't include it because they're installed directly onto VPSes that already have their own user management (thus you can just SFTP into the VPS itself in most cases).

**Reply by SpeedyCharly** [reply]
My friend was only given access to the Azuracast Dashboard and no access to the VPS itself... Guess he's up sh*t creek without a paddle...

**Reply by BusterNeece** [reply]
The footer should indicate what installation type it is. If it's a Docker install, SFTP is included out of the box, and if the reseller is turning it off, that's on their end. Ansible installations, the reseller would need to negotiate that separately and probably wouldn't give out SFTP access on account of it likely providing system-wide control.

**Reply by SpeedyCharly** [reply]
With my Docker setup I can SFTP to 2022 and that gives me restricted access to just the media folder and playlists.  I happen to have full access to the Admin side of Azuracast, but I haven't found anything that would allow me to turn SFTP access on or off (short of deleting the user I guess).  I have also full root access to the vps because I am self hosted but I don't need to use that to manage media content. I don't quite understand why the same restricted access couldn't be granted on Ansible.  

At this late hour, my buddy is in bed and I'll have to wait 2morrow to check his install method...

**Reply by BusterNeece** [reply]
Simply put, we just don't ship the sftpgo service on Ansible because if you're on Ansible, your VPS already has an SFTP service (its own built-in SSH server), so there's no need for us to implement that utility, as you don't have the same abstract filesystem that Docker has. It's a decision on our part, but also an issue with the reseller here if they're not using Docker installs or otherwise disabling SFTP.

**Reply by SpeedyCharly** [reply]
Ok thanks... that tells me where I should push to get results...  Have a good nite!

**Reply by SpeedyCharly** [reply]
Got around to checking and you were right, he's on Ansible...  So I guess it would be a matter of educating the reseller for him to grant restricted SFTP access...
